assets/js/react/GameStateEditor/components/EntityForm/EntityForm.jsx

```diff
@@ -46,132 +51,249 @@ class EntityForm extends Component {
     }
-    this.state = { entity: this.props.entity || { props: {} } }
+  }
+}
+
+class _EntityForm extends Component {
+  constructor(props) {
+    super(props)
+    this.state = emptyEntity
   }
 
   componentDidMount() {
     if (isEmpty(this.props.availableBehaviors)) {
       this.props.fetchAvailableBehaviors()
     }
+    if (!isEmpty(this.props.entity)) {
+      this.setState({ ...this.props.entity })
+    }
   }
 
-  handleChangeBehaviors = behaviors => {
+  handleRemoveBehavior = ev => {
     const currentState = this.state
     this.setState({
-      entity: {
-        ...currentState.entity,
-        behaviors: map(prop("value"), behaviors),
-      },
+      behaviors: omit([ev.target.name], this.state.behaviors),
     })
   }
 
-  handleChangePropsDropdown = fieldName => {
-    return selectedValue => {
-      const currentState = this.state
-      this.setState({
-        entity: {
-          ...currentState.entity,
-          props: {
-            ...currentState.entity.props,
-            [fieldName]: selectedValue.value,
-          },
-        },
+  handleAddNewBehavior = selectedOption => {
+    const currentState = this.state
+    this.setState(
+      mergeDeepRight(currentState, {
+        behaviors: { [selectedOption.value]: {} },
       })
-    }
+    )
   }
 
+  handleChangeLocation = selectedOption =>
+    this.setFieldState("props.location", selectedOption.value)
+
   handleChangeName = ev => {
     const currentState = this.state
     const newName = ev.target.value || ""
-    this.setState({
-      entity: {
-        ...currentState.entity,
+    console.log("change name", ev, currentState, newName)
+    this.setState(
+      mergeDeepRight(currentState, {
         id: newName
           .trim()
           .replace(/[^\w\d]+/gi, "-")
           .toLowerCase(),
         name: newName,
-      },
-    })
+      })
+    )
   }
 
-  handleChange = ev => {
-    const currentState = this.state
-    this.setState({
-      entity: {
-        ...currentState.entity,
-        [ev.target.name]: ev.target.value,
-      },
-    })
+  getFieldState = fieldName => {
+    return get(this.state, fieldName)
   }
 
-  handleChangeProps = ev => {
+  setFieldState = (fieldName, value) => {
     const currentState = this.state
-    this.setState({
-      entity: {
-        ...currentState.entity,
-        props: {
-          ...currentState.entity.props,
-          [ev.target.name]: ev.target.value,
-        },
-      },
-    })
+    const nameBits = fieldName.split(".")
+    const update = assocPath(nameBits, value)({})
+    this.setState(mergeDeepRight(currentState, update))
   }
 
+  handleChange = ev => this.setFieldState(ev.target.name, ev.target.value)
+
+  handleChangeDestination = fieldName => selectedOption =>
+    this.setFieldState(fieldName, selectedOption.value)
+
   handleAdd = ev => {
     ev.preventDefault()
-    this.props.addEntity(this.state.entity)
+    this.props.addEntity(this.state)
```

Comment:
the `addEntitySaga` is listening for `ADD_ENTITY` which is this action `addEntity`.  It calls the server with the new entity data and adds it to the world